server/app/mutations/stacks/common.rb

```diff
@@ -35,16 +35,11 @@ def validate_expose
     end
 
     # @param [Hash] service
-    def validate_service_links(service)
+    # @return [Array<Hash>] links to external services
+    def select_external_service_links(service)
       links = service[:links] || []
-      internal_links = links.select{ |l| !l['name'].include?('/') }
-      links = links - internal_links
-      internal_links.each do |l|
-        unless self.services.any?{|s| s[:name] == l['name']}
-          add_service_error(service[:name], :links, :exist,  "Linked service '#{l['name']}' does not exist")
-        end
-      end
-      service[:links] = links
```

Comment:
This kind of hidden mutation in a `validate` method also happens in the `GridServices::Common#validate_links`, which mutates the `link[:name]` to add the `stack/` prefix if missing. I think that ought to be fixed too.